pkg/interface/src/components/Omnibox.js

```diff
@@ -0,0 +1,254 @@
+        ...flattenedResults.filter((e) => {
+          return e.link === current;
+        })
+      );
+      if (currentIndex > 0) {
+      const nextLink = flattenedResults[currentIndex - 1].link;
+      this.setState({ selected: nextLink });
+      } else {
+      const nextLink = flattenedResults[totalLength - 1].link;
+      this.setState({ selected: nextLink });
+      }
+    } else {
+      const nextLink = flattenedResults[totalLength - 1].link;
+      this.setState({ selected: nextLink });
+    }
+  }
+
+  setNextSelected() {
+    const current = this.state.selected;
+    const flattenedResults = Array.from(this.state.results.values()).flat();
+    if (current !== '') {
+      const currentIndex = flattenedResults.indexOf(
+        ...flattenedResults.filter((e) => {
+          return e.link === current;
+        })
+      );
+      if (currentIndex < flattenedResults.length - 1) {
+      const nextLink = flattenedResults[currentIndex + 1].link;
+      this.setState({ selected: nextLink });
+      } else {
+      const nextLink = flattenedResults[0].link;
+      this.setState({ selected: nextLink });
+      }
+    } else {
+      const nextLink = flattenedResults[0].link;
+      this.setState({ selected: nextLink });
+    }
+  }
+
+  render() {
+    const { props, state } = this;
+    const categoryResult = [];
+
+    const renderResults = <Box maxHeight="400px" overflowY="scroll">
+      {categoryResult}
+    </Box>;
+
+      ['commands', 'subscriptions', 'groups', 'apps'].map((category, i) => {
+      const categoryResults = state.results.get(category);
+      if (categoryResults.length > 0) {
+        const each = categoryResults.map((result, i) => {
+          return <OmniboxResult
+            key={i}
+            icon={result.app}
+            text={result.title}
+            subtext={cite(result.host)}
+            link={result.link}
+            navigate={() => this.navigate(result.link)}
+            selected={this.state.selected}
+            dark={props.dark}
+                 />;
+        });
+         categoryResult.push(<Box key={i} width='max(50vw, 300px)' maxWidth='600px'>
+        <Rule borderTopWidth="0.5px" color="washedGray" />
+        <Text gray ml={2}>{category.charAt(0).toUpperCase() + category.slice(1)}</Text>
+          {each}
+        </Box>);
+      }
+    });
+
+    return (
+      <>
+        <Box
+          backgroundColor='lightGray'
+          width='100vw'
+          height='100vh'
+          position='absolute'
+          top='0'
+          right='0'
+          zIndex='9'
+          display={props.show ? 'block' : 'none'}
+        >
+          <Row justifyContent='center'>
+            <Box
+              mt='20vh'
+              width='max(50vw, 300px)'
+              maxWidth='600px'
+              borderRadius='2'
+              backgroundColor='white'
+              ref={(el) => {
+                this.omniBox = el;
+              }}
+            >
+              <input
```

Comment:
I think this input should be made into its own component to better separate concerns of this file.

Reply:
Are we able to retain the ref if it's a separate component?